javagen/src/main/java/com/azure/autorest/template/WrapperClientMethodTemplate.java

```diff
@@ -55,6 +55,10 @@ public void write(ClientMethod clientMethod, JavaType typeBlock) {
         if (clientMethod.getType().name().contains("Async") && methodName.endsWith("Async")) {
             methodName = methodName.substring(0, methodName.length() - "Async".length());
         }
+
+        if (clientMethod.getType().name().contains("Sync")) {
+            methodName = methodName.replace("Sync", "");
+        }
```

Comment:
That seems to be the wrong place of the solution? I would assume whether or not "sync-stack" is enabled, the name of client method stays the same (without "Sync" in it) even in implementation. <-- why should method name expose its implementation detail?
@alzimmermsft personally I feel this to be a bug in ClientTemplateMapper when "sync-stack" enabled.

Reply:
It is a bit related to the method naming, given now that the name of the proxy method could be different in sync mode.
https://github.com/Azure/autorest.java/pull/1794#discussion_r1000118501

Reply:
Agreed, I don't think `sync-stack` should be adding a suffix of `Sync` to synchronous methods. This wasn't being done before for synchronous methods and may end up being a breaking change in a lot of places.